core/network/src/main/java/com/gkp/core/network/AccessTokenRefreshHandler.kt

```diff
@@ -0,0 +1,43 @@
+package com.gkp.core.network
+
+import com.gkp.auth.domain.session.SessionStorage
+import com.gkp.core.network.model.RefreshTokenBody
+import com.gkp.core.network.util.networkApiCall
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.runBlocking
+import okhttp3.Response
+import java.net.HttpURLConnection.HTTP_UNAUTHORIZED
+
+class AccessTokenRefreshHandler(
+    private val sessionStorage: SessionStorage,
+    private val taskyRetrofitApi: TaskyRetrofitApi
+) {
+
+    fun refreshAccessToken() {
+        runBlocking {
+            val authInfo = sessionStorage.getAuthInfo()
+            networkApiCall {
+                val refreshTokenResponse = taskyRetrofitApi.refreshToken(
+                    refreshTokenBody = RefreshTokenBody(
+                        refreshToken = authInfo.refreshToken,
+                        userId = authInfo.userId
+                    )
+                )
+                sessionStorage.saveAuthInfo(
+                    authInfo.copy(
+                        accessToken = refreshTokenResponse.accessToken,
+                        accessTokenExpirationTimestamp = refreshTokenResponse.accessTokenExpirationTimestamp
+                    )
+                )
+            }.launchIn(this)
+        }
+    }
```

Comment:
This looks very dangerous with runBlocking. It is safe when executed from a background thread, like in the interceptor, but in that case I'd let the interceptor use runBlocking instead and not this function (make this suspend instead). That way, you avoid accidentally calling this function from somewhere else where you might accidentally freeze the main thread. Exposing it as suspend delegates the responsibility to the caller instead, which is generally desirable.